### DISCUSSION TITLE
Autotype and window focus

### DISCUSSION THREAD

**Discussion by silicispongia** [keepassxreboot/keepassxc#10344] [Q&A] [answered]
Checking in to see if this is just me, and/or there is a setting I can switch or other workaround to address this behavior.

I'm running KeepassXC for MacOS. (Sonoma version, but this has been happening for as long as I can remember)
"Always ask before performing Auto-Type" is enabled.
When I press the Global Auto-Type shortcut, as configured in KeepassXC, KeepassXC responds as expected, but does not grab focus.

The current way it works:
1. press universal hotkey while at a password prompt. (Web browser, terminal, and other authentication dialogs)
2. KeepassXC creates a window asking which password/autotype sequence I want to use
3. The password field is still in focus, so any further keyboard input is interpreted there.
4. grab my mouse and click on that window, sometimes it takes a few clicks
5. click into the search box and make sure a cursor is displayed
6. search for the entry i want
7. highlight the entry with the arrow keys if necessary
8. press enter
9. the autotype works as expected and types in the correct window.

It seems to only happen when entering passwords. Normal text entry works as expected. Is there some focus-stealing protection in MacOS that we're running afoul of?

**Comment by droidmonkey** [comment, marked as answer]
MacOS purposefully doesn't allow focus grab from password fields. You need to manually shift focus using the mouse

**Comment by silicispongia** [comment]
I was afraid it was something like that. I bet they have an exception for their own password manager. Jerks
________________________________
